frontend/packages/topology/src/components/export-app/ExportViewLogButton.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { Button, Tooltip } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { resourcePath, history } from '@console/internal/components/utils';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { JobModel, PodModel } from '@console/internal/models';
+import { PodKind, JobKind } from '@console/internal/module/k8s';
+import { isModifiedEvent } from '@console/shared/src';
+import { EXPORT_JOB_NAME } from '../../const';
+
+interface ExportViewLogButtonProps {
+  namespace: string;
+  onViewLog?: () => void;
+}
+
+const ExportViewLogButton: React.FC<ExportViewLogButtonProps> = ({ namespace, onViewLog }) => {
+  const { t } = useTranslation();
+  const [job, jobLoaded] = useK8sWatchResource<JobKind>({
+    kind: JobModel.kind,
+    name: EXPORT_JOB_NAME,
+    namespace,
+    isList: false,
+  });
+
+  const podResource = React.useMemo(
+    () =>
+      jobLoaded &&
+      job?.metadata && {
+        kind: PodModel.kind,
+        isList: false,
+        namespace: job.metadata.namespace,
+        selector: job.spec.selector,
+      },
+    [job, jobLoaded],
+  );
+
+  const [podData, podLoaded] = useK8sWatchResource<PodKind>(podResource);
```

Comment:
Be careful when using ternary operators. In this case `useK8sWatchResource` expects param of type `WatchK8sResource | null` but `podResource` doesn't match this type. Looks like typescript type inference doesn't properly acknowledge that `podResource` may be `false` due to `jobLoaded`.

Reply:
Yeah, explicit `job && jobLoaded ? { ... } : null` would make this more typesafe.

Reply:
returning null incase object cannot be formed